plugins/test/framework.h

```diff
@@ -28,6 +28,42 @@
 
 namespace service_extensions_samples {
 
+class Buffer : public proxy_wasm::BufferBase {
+ public:
+  Buffer() = default;
+
+  // proxy_wasm::BufferInterface
+  size_t size() const override;
+  proxy_wasm::WasmResult copyTo(proxy_wasm::WasmBase* wasm, size_t start,
+                                size_t length, uint64_t ptr_ptr,
+                                uint64_t size_ptr) const override;
+  proxy_wasm::WasmResult copyFrom(size_t start, size_t length,
+                                  std::string_view data) override;
+
+  // proxy_wasm::BufferBase
+  void clear() override {
+    proxy_wasm::BufferBase::clear();
+    string_buffer_ = nullptr;
+  }
+  Buffer* set(std::string_view data) {
+    return static_cast<Buffer*>(proxy_wasm::BufferBase::set(data));
+  }
+  Buffer* set(std::unique_ptr<char[]> owned_data, uint32_t owned_data_size) {
+    return static_cast<Buffer*>(
+        proxy_wasm::BufferBase::set(std::move(owned_data), owned_data_size));
+  }
+
+  Buffer* set(std::string* string) {
+    clear();
+    string_buffer_ = string;
+    return this;
+  }
+  std::string get() { return *string_buffer_; }
+
+ private:
+  std::string* string_buffer_;
```

Comment:
Is this string owned by Buffer or not? Could the existing `data_` or `owned_data_` fields be used instead?